verify.go

```diff
@@ -6,6 +6,24 @@ import (
 	"time"
 )
 
+// VerifyRange verifies untrusted Headers against trusted following general Header checks and
+// custom user-specific checks defined in [Verify].
+//
+// In case of error, returns verified sub-range of Headers and error.
+func VerifyRange[H Header[H]](trstd H, untrstd []H) ([]H, error) {
```

Comment:
Despite simple logic this should be covered by unit-tests. Probably `headertest` should have it to prevent import cycle.